pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -0,0 +1,397 @@
+package ibmcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/IBM-Cloud/bluemix-go/api/cis/cisv1"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/controllerv2"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/managementv2"
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/bluemix-go/models"
+	"github.com/IBM-Cloud/bluemix-go/session"
+	"github.com/IBM/go-sdk-core/v4/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	ibmcloudtypes "github.com/openshift/installer/pkg/types/ibmcloud"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockgen -source=./client.go -destination=./mock/ibmcloudclient_generated.go -package=mock
+
+// API represents the calls made to the API.
+type API interface {
+	GetCISInstance(ctx context.Context, crnstr string) (*models.ServiceInstanceV2, error)
+	GetCustomImageByName(ctx context.Context, imageName string, region string) (*vpcv1.Image, error)
+	GetCustomImages(ctx context.Context, region string) ([]vpcv1.Image, error)
+	GetDNSZones(ctx context.Context) ([]ibmcloudtypes.DNSZoneResponse, error)
+	GetEncryptionKey(ctx context.Context, keyCRN string) (*ibmcloudtypes.EncryptionKeyResponse, error)
+	GetResourceGroups(ctx context.Context) ([]models.ResourceGroupv2, error)
+	GetResourceGroup(ctx context.Context, nameOrID string) (*models.ResourceGroupv2, error)
+	GetSubnet(ctx context.Context, subnetID string) (*vpcv1.Subnet, error)
+	GetVSIProfiles(ctx context.Context) ([]vpcv1.InstanceProfile, error)
+	GetVPC(ctx context.Context, vpcID string) (*vpcv1.VPC, error)
+	GetVPCZonesForRegion(ctx context.Context, region string) ([]string, error)
+	GetZoneIDByName(ctx context.Context, crn string, name string) (string, error)
+}
+
+// Client makes calls to the IBM Cloud API.
+type Client struct {
+	ssn           *session.Session
+	managementAPI managementv2.ResourceManagementAPIv2
+	controllerAPI controllerv2.ResourceControllerAPIV2
+	cisAPI        cisv1.CisServiceAPI
+	vpcAPI        *vpcv1.VpcV1
+}
+
+// cisServiceID is the Cloud Internet Services' catalog service ID.
+var cisServiceID = "75874a60-cb12-11e7-948e-37ac098eb1b9"
+
+// NewClient initializes a client with a session.
+func NewClient(ctx context.Context) (*Client, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

Comment:
Why go through the effort of setting a timeout on the context, when the context is not even used? I suggest removing the context from the parameters to `NewClient`.